Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -21,7 +21,10 @@ open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType
 
         // Creates an observable that starts a request each time it's subscribed to.
         return Observable.create { [weak self] observer in
```

Comment:
What do you think @ashfurrow? I think you did a lot of the early work here. What makes more sense to you: Having the `Observable` be usable regardless of what happens with the `RxMoyaProvider` that generated it or emitting an error if that provider gets released? Maybe we can also think about how that would be consistent with behavior of `MoyaProvider`. If you release a `MoyaProvider` will inflight requests still be finished?

Also, to note, if we adopt the pattern of emitting an error on release, I think users might still be able to work around this by retaining the provider inside of a closure retained by the `Observable` (e.g. `do` or `subscribe`). A bit hacky though...